app.py

```diff
@@ -181,6 +198,38 @@ def azul_match_alias(index, alias, from_=None, size=10):
     return es_query(index=index, query=dsl, size=size)
 
 
+def azul_get_document(key, val, name, es_index, map_fn):
+    """
+    Queries ElasticSearch for a single document and returns a
+    :class:`~chalice.Response` object with the retrieved data. Wrapper
+    around :func:`azul_match_field`. Implements lookup functionality used
+    in :func:`get_data_object` and :func:`get_data_bundle`.
+    :param str key: the key to search for in the given ElasticSearch index
+    :param str val: the value to search for in the given ElasticSearch index
+    :param str name: the key the document should be returned under
+    :param str es_index: the name of the index to query in ElasticSearch
+    :param callable map_fn: function mapping the returned Azul document to a
+                            DOS format
+    :raises RuntimeError: if the ElasticSearch response is not understood
+    :rvtype: :class:`chalice.Response`
+    :returns: the retrieved data or the error state
+    """
+    try:
+        data = azul_match_field(index=es_index, key=key, val=val)
+        data = map_fn(data)
+        # Double check to verify identity
+        if data['id'] != val:
+            raise LookupError("ID mismatch in results")
+    # azul_match_field will also raise a LookupError if no results are returned
+    except LookupError:
+        return Response({'msg': "No results found for type %s and ID %s." % (name, val)},
+                        status_code=404)
+    # es_query will raise RuntimeError if it doesn't understand the ES response
+    except RuntimeError:
+        return Response({'msg': "Unexpected response from Azul."}, status_code=500)
+    return Response({name: data}, status_code=200)
+
```

Comment:
When exceptions occur, they should be logged, so the exceptions are present in the Lambda CloudWatch logs for troubleshooting purposes (and monitoring/altering, etc.). Unless these are already being logged adequately by some of the lower layers, yet because you are not using client libraries (e.g. https://github.com/elastic/elasticsearch-py) I suspect they may not be. The extent to which appropriate logging is occurring (or not) is one of the things I had hoped/intended to investigate by running the tests locally on my system.